### PR TITLE
added tests for fromHeader and fromParams in request authentication

### PR DESCRIPTION
**Pull request by aryan16**
**Please provide a description of this PR:**

Added additional tests in request authentication. Added jwt token in  `fromHeaders` and `fromParams` and tested it against jwksuri.

**To help us figure out who should review this PR, please put an X in all the areas that this PR affects.**

- [ ] Configuration Infrastructure
- [ ] Docs
- [ ] Installation
- [ ] Networking
- [ ] Performance and Scalability
- [ ] Policies and Telemetry
- [X] Security
- [ ] Test and Release
- [ ] User Experience
- [ ] Developer Infrastructure

**Please check any characteristics that apply to this pull request.**

- [X] Does not have any [user-facing](https://github.com/istio/istio/tree/master/releasenotes#when-to-add-release-notes) changes. This may include CLI changes, API changes, behavior changes, performance improvements, etc.

### CODE REVIEW

**Comment by aryan16** [comment]
cc : @yangminzhu

File: tests/integration/security/jwt_test.go [hunk 1]
```diff
@@ -309,6 +309,78 @@ func TestRequestAuthentication(t *testing.T) {
...
+						},
+						ExpectResponseCode: response.StatusCodeOK,
+					},
+					{
+						Name:   "valid-params-secondary",
+						Config: "headers-params",
+						CallOpts: echo.CallOptions{
+							PortName: "http",
+							Scheme:   scheme.HTTP,
+							Path:     "/valid-token?secondary_token=" + jwt.TokenIssuer1,
```
Comment (howardjohn) [inline tests/integration/security/jwt_test.go:329]: can you add a test for `secondary_token` and `token` set

Comment (aryan16) [reply tests/integration/security/jwt_test.go:329]: oh you mean when both the tokens passed in the same query params together ?

File: tests/integration/security/jwt_test.go [hunk 2]
```diff
@@ -309,6 +309,78 @@ func TestRequestAuthentication(t *testing.T) {
...
+						ExpectResponseCode: response.StatusCodeForbidden,
+					},
+					{
+						Name:   "valid-header",
+						Config: "headers-params",
+						CallOpts: echo.CallOptions{
+							PortName: "http",
+							Scheme:   scheme.HTTP,
+							Headers: map[string][]string{
+								"X-Jwt-Token": {"Value " + jwt.TokenIssuer1},
```
Comment (yangminzhu) [inline tests/integration/security/jwt_test.go:352]: hmm, how is this valid without using the prefix `Bearer `?

Comment (aryan16) [reply tests/integration/security/jwt_test.go:352]: oh I mistakenly added the prefix `Bearer` but it is strange to see that the test case passed. I will look into it.

**Comment by aryan16** [comment]
cc : @yangminzhu

File: tests/integration/security/jwt_test.go (hunk 1 above)
Comment (howardjohn) [reply tests/integration/security/jwt_test.go:329]: yes

**Review by howardjohn — APPROVED** [review]

**Pull request merged**